test/test_intensify/test_successive_halving.py

```diff
@@ -1461,6 +1461,227 @@ def target(x):
         # For all runs to be completed before moving to the next stage
         self.assertEqual(intent, RunInfoIntent.WAIT)
 
+    def _exhaust_stage_execution(self, intensifier, taf, challengers):
+        """
+        Exhaust configuration/instances seed and returns the
+        run_info that were not launched.
+
+        The idea with this procedure is to emulate the fact that some
+        configurations will finish while others won't. We need to be
+        robust against this scenario
+        """
+        pending_processing = []
+        toggle = True
```

Comment:
I think it would be great if the toggle could be somehow randomized. I guess we know how many runs are generated, so we could sample a boolean toggle mask to use below.